### DISCUSSION TITLE
How to integrate tailwindcss use

### DISCUSSION THREAD

**Discussion by vaynevayne** [emotion-js/emotion#3045] [Q&A] [answered]
I'm going to use css in js, but don't want to write the style myself, and instead use the class name of tailwindcss directly

**Comment by Menthol2k** [comment, marked as answer]
1. Install Tailwind CSS Start by installing Tailwind CSS in your project. You can use npm 
 `npm install tailwindcss`
2. Configure Tailwind CSS: After installing, you'll need to generate a Tailwind CSS configuration file. 
`npx tailwindcss init`

3. Open the tailwind.config.js file and customize it according to your requirements. You can modify colors, spacing, breakpoints, and other aspects of the generated CSS classes.

4. Import Tailwind CSS: In your CSS-in-JS solution (such as styled-components, Emotion, or any other library), import the Tailwind CSS styles by adding the following line at the top of your file

`import 'tailwindcss/tailwind.css';`

5. This will import the pre-generated CSS file that contains all the Tailwind CSS utility classes.

Use Tailwind CSS classes: Now, you can use Tailwind CSS class names directly in your CSS-in-JS solution by applying the desired classes to your components. 
I can give you an example:

```
import styled from 'styled-components';

const Button = styled.button`
  /* Apply Tailwind CSS classes */
  ${tw`px-4 py-2 bg-blue-500 text-white rounded`}
`;
```

In the above example, the tw template literal is a utility provided by Tailwind CSS that allows you to apply Tailwind CSS class names within the CSS-in-JS template string.

By following these steps, you can leverage the power of Tailwind CSS utility classes without having to write the styles manually.

**Reply by vaynevayne** [reply]
That's just what I needed. Thank you

**Comment by vaynevayne** [comment]
@Menthol2k  Hello, is there documentation for tw template strings?
